discovery-infra/tests/base_cancel_reset.py

```diff
@@ -0,0 +1,43 @@
+import os
+import pytest
+
+import assisted_service_api
+import utils
+import consts
+
+if os.environ.get('NODE_ENV') == 'QE_VM':
+    #TODO change import to "as"
+    import node_controlers.qe_vm_controler 
+
+ssh_public_key = os.environ.get('SSH_PUB_KEY')
+remote_service_url = os.environ.get('REMOTE_SERVICE_URL')
+pull_secret = os.environ.get('PULL_SECRET')
+cluster_name = os.environ.get('CLUSTER_NAME')
+offline_token = os.environ.get('OFFLINE_TOKEN')
+openshift_version = os.environ.get('OPENSHIFT_VERSION')
+base_domain = os.environ.get('BASE_DOMAIN')
+iso_download_path = os.environ.get('ISO')
+num_masters = int(os.environ.get('NUM_MASTERS'))
+num_workers = int(os.environ.get('NUM_WORKERS'))
+
+nodes_count = num_masters + num_workers
+
+class BaseCancelReset(object):
+    @pytest.fixture
+    def node_controler(self):
```

Comment:
There's no need to create class for the fixtures. They can be public.
IMO, it's best practice to import everything on top of the module instead of inside functions.
```python
from assisted_service_api import create_client
from node_controlers.qe_vm_controler import QeVMController

@pytest.fixture
def node_controler():
    if os.environ.get('NODE_ENV') == 'QE_VM':
        yield QeVMController()
    else:
        raise NotImplementedError()

@pytest.fixture
def api_client():
    yield create_client()
```

Reply:
The class (BaseTest) now holds the fixtures, and contains helper functions so that any new test plan can be a class that inherits from BaseTest